drasyl-extras/src/main/java/org/drasyl/handler/connection/OutgoingSegmentQueue.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2020-2022 Heiko Bornholdt and Kevin Röbert
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+ * OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.drasyl.handler.connection;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import org.drasyl.util.logging.Logger;
+import org.drasyl.util.logging.LoggerFactory;
+
+import java.util.ArrayDeque;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents the outgoing segment queue that holds segments to be sent over a connection.
+ * <p>
+ * This class manages the queue of segments that are waiting to be sent over a connection. The queue
+ * implements a first-in, first-out (FIFO) ordering, where segments are added to the end of the
+ * queue and retrieved from the front of the queue. The queue is used by the sender to hold segments
+ * that are waiting to be sent to the receiver. Segments are added to the queue as they are
+ * generated by the application, and they are sent out by the sender according to the congestion
+ * control algorithm and the available network resources.
+ */
+public class OutgoingSegmentQueue {
+    private static final Logger LOG = LoggerFactory.getLogger(OutgoingSegmentQueue.class);
+    private final ArrayDeque<Object> queue;
+
+    OutgoingSegmentQueue(final ArrayDeque<Object> queue) {
+        this.queue = requireNonNull(queue);
+    }
+
+    OutgoingSegmentQueue() {
+        this(new ArrayDeque<>());
+    }
+
+    void place(final ChannelHandlerContext ctx, final Segment seg, final ChannelPromise promise) {
```

Comment:
Warum heißen deine Methoden `place` und `flush` und nicht `enqueue` und `dequeue`

Reply:
Habs nun auch `add` genannt. Ich bleibe aber lieber bei `flush`, weil wir wirklich in den Channel rein flushen